bin/ttl-to-json

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env node
+
+// Converts a TTL RDF file to JSON or JSON-LD. 
+
+var _ = require('underscore');
+var cmdlineArgs = require('command-line-args')
+var fs = require('fs');
+var path = require('path');
+var util = require('util');
+
+var Promise = require('promise');
+var rdfstore = require('rdfstore');
+var jsonld = require('jsonld').promises;
+
+/**
+ * Converts the given graph object into a JSON LD Graph object using an optional
+ * frame on the corresponding Component.
+ *
+ * @param input RDF JS Interface Graph object
+ * @param frame JSON-LD Frame object
+ *
+ * @see http://json-ld.org/spec/latest/json-ld-framing/
+ */
+function rdfToJsonLd(input, frame) {
+
+    return buildJSON(input).then(function(json) {
+
+        if (frame) {
+            return new Promise(function(resolve) { 
+                var framedJson = jsonld.frame(json, frame);
+                resolve(framedJson); 
+            }).then(framedJson => { 
+
+                return filterBNodes(framedJson); 
+
+            }).catch(function(e) {
+                console.error("Unable to process JSONLD frame!");
+                console.error(e);
+                process.exit(1);
+            });
+
+       } else {
+
+            json = filterBNodes(json);
+            if (json.length == 1) 
+               json[0];
+
+           return json;
+       }
+    });
+}
+
+/**
+ * Walks the graph looking for blank node IDs that were added by the jsonld library.  Blank
+ * node IDs that are referenced only once are removed.  This is done to ensure we return
+ * to the original JSON-LD input without any RDF remnants
+ * 
+ * @param graph - a JSON or JSON-LD graph
+ *
+ * @return the graph will be updated to remove RDF ID blank nodes 
+ */
+function filterBNodes(graph) { 
+
+    // Generate a list of blank nodes, with a count of the number of times referenced
+    // and an array that has the path to the first blank node reference.
+    function bNodeHash(graph, bNodes, jspath) { 
```

Comment:
Does this function need to be nested?  It makes the outer function harder to read.